File: 01252.py
a, b = map(int, input().split())

a = str(a); b = str(b)

if len(a) >= len(b): up = a; down = b
else: up = b; down = a

length = len(down)
up_index = len(up) - 1
down_index = len(down) - 1

carry = 0; result = ""
for i in range(length):
    tmp = int(up[up_index]) + int(down[down_index]) + carry

    carry = tmp // 2
    result = str(tmp % 2) + result

    up_index -= 1
    down_index -= 1

while up_index >= 0:
    tmp = int(up[up_index]) + carry

    carry = tmp // 2
    result = str(tmp % 2) + result

    up_index -= 1

if carry: result = str(carry) + result

print(result)

File: 01252.cpp
#include <iostream>
using namespace std;

int main() {
	string a, b; cin >> a >> b;

	int index = 0;
	while (a[index++] == '0');
	a = a.substr(--index);
	if (a == "") a = "0";
	
	index = 0;
	while (b[index++] == '0');
	b = b.substr(--index);
	if (b == "") b = "0";

	string up, down;
	if (a.length() >= b.length()) {
		up = a; down = b;
	}
	else {
		up = b; down = a;
	}

	int len = down.length();
	int up_index = up.length() - 1;
	int down_index = down.length() - 1;

	int carry = 0; string result = "";
	for (int i = 0; i < len; i++) {
		int tmp = (up[up_index] - '0') + (down[down_index] - '0') + carry;

		carry = tmp / 2;
		result = (char)(tmp % 2 + '0') + result;

		up_index--; down_index--;
	}

	for (; up_index >= 0; up_index--) {
		int tmp = (up[up_index] - '0') + carry;

		carry = tmp / 2;
		result = (char)(tmp % 2 + '0') + result;
	}

	if (carry) result = (char)(carry + '0') + result;
	// if (result[0] == '0') result = "0";

	cout << result << endl;
}
